spec/services/google/scrape_service_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Google::ScrapeService, type: :service do
+  describe '#call!' do
+    context 'given a VALID keyword_id' do
+      context 'given NO errors' do
+        it 'saves the html of the result page', vcr: 'google/google_no_ads' do
+          keyword = Fabricate :keyword
+
+          described_class.new(keyword_id: keyword.id).call!
+          updated_keyword = keyword.reload
+
+          expect(updated_keyword.result_page_html).to be_a String
+        end
+
+        it 'saves the links count', vcr: 'google/google_no_ads' do
+          keyword = Fabricate :keyword
+
+          described_class.new(keyword_id: keyword.id).call!
+          updated_keyword = keyword.reload
+
+          expect(updated_keyword.links_count).to be_a Integer
+        end
+
+        it 'saves all search entires', vcr: 'google/google_no_ads' do
```

Comment:
should we test it updates the status to `succeed`?

Reply:
Fixed in b702425f6dac9764fb1c4a4c391774acddc991a1